universum/configuration_support.py

```diff
@@ -1,19 +1,286 @@
+        self.artifacts: str = artifacts
+        self.report_artifacts: str = report_artifacts
+        self.artifact_prebuild_clean: bool = artifact_prebuild_clean
+        self.directory: str = directory
+        self.critical: bool = critical
+        self.background: bool = background
+        self.finish_background: bool = finish_background
+        self.code_report: bool = code_report
+        self.pass_tag: str = pass_tag
+        self.fail_tag: str = fail_tag
+        self.if_env_set: str = if_env_set
+        self.skip_numbering_level: bool = False
+        self.children: Optional['Variations'] = None
+        self.extras: Dict[str, str] = extras if extras else {}
+        for key, value in kwargs.items():
+            self.extras[key] = value
+
+    def replace_string(self, from_string: str, to_string: str) -> None:
+        self.command = [word.replace(from_string, to_string) for word in self.command]
+        self.artifacts = self.artifacts.replace(from_string, to_string)
+        self.report_artifacts = self.report_artifacts.replace(from_string, to_string)
+        self.directory = self.directory.replace(from_string, to_string)
+        # should we update extras? (user-defined)
+
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, dict):
+            for key, val in other.items():
+                if getattr(self, key, None) != val:
+                    return False
+            return True
+        return super().__eq__(other)
+
+    def __getitem__(self, item: str) -> Optional[str]:
+        return self.extras.get(item, None)
+
+    def __add__(self, other: 'ProjectConfiguration') -> 'ProjectConfiguration':
+        return ProjectConfiguration(
+            name=self.name + other.name,
+            command=self.command + other.command,
+            environment={**self.extras, **other.extras},
+            artifacts=self.artifacts + other.artifacts,
+            report_artifacts=self.report_artifacts + other.report_artifacts,
+            artifact_prebuild_clean=self.artifact_prebuild_clean or other.artifact_prebuild_clean,
+            directory=self.directory + other.directory,
+            background=self.background or other.background,
+            finish_background=self.finish_background or other.finish_background,
+            code_report=self.code_report or other.code_report,
+            pass_tag=self.pass_tag + other.pass_tag,
+            fail_tag=self.fail_tag + other.fail_tag,
+            if_env_set=self.if_env_set + other.if_env_set,
+            extras={**self.extras, **other.extras}
+        )
+
+    def stringify_command(self) -> bool:
+        result = False
+        command_line = ""
+        for argument in self.command:
+            if " " in argument:
+                argument = "\"" + argument + "\""
+                result = True
+            command_line = command_line + " " if command_line else "" + argument
+        self.command = [command_line]
```

Comment:
self.comman is List of strings, but this used to assign one string. Change to Union or report as list of 1 element?